enhancements/baremetal/enable-baremetal-on-other-platforms.md

```diff
@@ -0,0 +1,198 @@
+
+### Non-Goals
+
+Allow Baremetal Host API to be fully enabled on all platforms.
+
+Allow Machine API integration across platform types.
+- The centralized host management flow currently interacts directly via this API
+  without any Machine API integration.
+- This means further work will be required via a future enhancement to enable the
+  single cluster case where a combination of e.g VM controlplane and Baremetal
+  workers is desired.
+
+## Proposal
+
+BMO (baremetal-operator) provides the Baremetal Host API, it in turn is configured
+and managed by CBO (cluster-baremetal-operator).
+
+CBO reads the Provisioning CR that is created by the installer on baremetal platforms
+and uses that to configure and deploy BMO. In the case of non-baremetal platforms
+the user (or automation) will need to define the Provisioning CR.
+
+Currently CBO checks the platform and if it is not baremetal it will be in a "disabled" state i.e. it will
+1. set status.conditions Disabled=true and
+2. not read or process the Provisioning CR and thus not deploy baremetal-operator.
+
+This proposal is to allow CBO to be enabled on the following platforms only:
+- Baremetal (current)
+- None
+- OpenStack
+- vSphere.
+
+Note: The Hypershift use case will be explicitly disallowed by disabling CBO when
+Infrastructure.Status.ControlPlaneTopology == "External".
+
+Further (to restrict the testing matrix) the allowed configuration options
+of the Provisioning CR will be restricted to exactly those required by centralized host management.
+
+*Only spec.provisioningNetwork=Disabled mode will be accepted in the Provisioning CR.*
+
+If any other provisioningNetwork mode is set, the CBO webhook will refuse the change
+in the usual case, but if defined before upgrading the operator, the Reconcile loop
+must always validate the Provisioning CR.
+
+Note:
+
+1. when the Provisioning CR is set to provisioningNetwork=Disabled mode, worker
+nodes would be booted via virtual media. This removes the requirement for the
+Provisioning Network which can be expected to be available only in Baremetal platform types.
+
+2. documentation will need to be added to the centralized host management documentation
+explaining how to create and update a Provisioning CR for these platforms.
+
+### User Stories
+
+#### Story 1 - Current IPI baremetal platform use case
+
+No change.
+
+#### Story 2 - centralized host management use case
+
+As a user of a hub cluster that performs central infrastructure management, and
+optionally zero-touch provisioning, I need to provision hosts using the k8s-native
+API (Baremetal Hosts CR) even when the hub cluster has a platform of None, OpenStack, or vSphere.
+
+
+### Risks and Mitigations
+
+There is concern that *random* customers will use this feature out of context
+and create support burden. This is why we have not suggested enabling CBO on
+all platforms and with full feature set. However it is still a potential issue.
```

Comment:
One mitigation for this we've discussed is to avoid documenting this outside of the CIM/ZTP case.  For that reason this change won't be documented as a standalone feature, only in the context of CIM/ZTP.